openedx/core/djangoapps/credit/tests/test_verification_access.py

```diff
@@ -0,0 +1,270 @@
+
+        # Update the verification partitions.
+        # The existing partitions should still be available
+        self._update_partitions()
+        partition_ids = [p.id for p in self.course.user_partitions]
+        self.assertEqual(len(partition_ids), 3)
+        self.assertIn(0, partition_ids)
+        self.assertIn(1, partition_ids)
+
+    def test_multiple_reverification_blocks(self):
+        # Add an additional ICRV block in another section
+        other_icrv = ItemFactory.create(parent=self.verticals[3], category='edx-reverification-block')
+        self._update_partitions()
+
+        # Expect that both ICRV blocks have corresponding partitions
+        self.assertEqual(len(self.course.user_partitions), 2)
+        partition_locations = [p.parameters.get("location") for p in self.course.user_partitions]
+        self.assertIn(unicode(self.icrv.location), partition_locations)
+        self.assertIn(unicode(other_icrv.location), partition_locations)
+
+        # Delete the first ICRV block and update partitions
+        icrv_location = self.icrv.location
+        self.store.delete_item(
+            self.icrv.location,
+            ModuleStoreEnum.UserID.test,
+            revision=ModuleStoreEnum.RevisionOption.published_only
+        )
+        self._update_partitions()
+
+        # Expect that the correct partition is marked as inactive
+        self.assertEqual(len(self.course.user_partitions), 2)
+        partitions_by_loc = {
+            p.parameters["location"]: p
+            for p in self.course.user_partitions
+        }
+        self.assertFalse(partitions_by_loc[unicode(icrv_location)].active)
+        self.assertTrue(partitions_by_loc[unicode(other_icrv.location)].active)
+
+    def test_query_counts_with_no_reverification_blocks(self):
+        # Delete the ICRV block, so the number of ICRV blocks is zero
+        self.store.delete_item(
+            self.icrv.location,
+            ModuleStoreEnum.UserID.test,
+            revision=ModuleStoreEnum.RevisionOption.published_only
+        )
+
+        # 2 calls: get the course (definitions + structures)
+        # 2 calls: look up ICRV blocks in the course (definitions + structures)
+        with check_mongo_calls_range(max_finds=4, max_sends=2):
+            self._update_partitions(reload_items=False)
+
+    def test_query_counts_with_one_reverification_block(self):
+        # One ICRV block created in the setup method
+        # Additional call to load the ICRV block
+        with check_mongo_calls_range(max_finds=5, max_sends=3):
+            self._update_partitions(reload_items=False)
+
+    def test_query_counts_with_multiple_reverification_blocks(self):
+        # Total of two ICRV blocks (one created in setup method)
+        # Additional call to load each ICRV block
+        ItemFactory.create(parent=self.verticals[3], category='edx-reverification-block')
+        with check_mongo_calls_range(max_finds=6, max_sends=3):
```

Comment:
Note to reviewers: This is one part of the PR that I'm concerned about.  The number of Mongo queries is increasing linearly with the number of ICRV blocks in the course.  This wouldn't be such a big deal except that `modulestore().get_items()` is finding "deleted" blocks that have been orphaned from the course tree.  So if a course author creates and deletes a bunch of ICRV blocks, we could see performance issues on publish in credit courses.  I'm working with the Platform team to figure out if there's a way to pre-fetch the field we need from the blocks in a single query.